stripped/colabs/Birch_stripped.py

```diff
@@ -0,0 +1,732 @@
+wyniki["Precision [%]"] = wyniki["Name"].apply(lambda name: round(precision_score(listed_of_merged_df[0]["pustynia_i_step"], predykcje_df[name])*100, 2))
+wyniki["Recall [%]"] = wyniki["Name"].apply(lambda name: round(recall_score(listed_of_merged_df[0]["pustynia_i_step"], predykcje_df[name])*100, 2))
+wyniki["F1-Score [%]"] = wyniki["Name"].apply(lambda name: round(f1_score(listed_of_merged_df[0]["pustynia_i_step"], predykcje_df[name])*100, 2))
+wyniki.sort_values(by=['Accuracy [%]'], ascending=False)
+
+
+# # Tunning hiperparametrów modelu trenowanego na danych z lipca 2023
+
+# In[19]:
+
+
+X = nasa_202307.loc[:,'Rainf':'SoilM_100_200cm']
+X_scaled = MinMaxScaling(StandardScaling(X, X.columns), X.columns)
+X_scaled = pd.concat([nasa_202307[["lon", "lat"]], X_scaled], axis=1)
+merged_df = X_scaled.merge(NASA_an[['lon', 'lat', 'pustynia', 'pustynia_i_step']], left_on=['lon','lat'], right_on = ['lon','lat'], how='inner')
+
+
+# In[20]:
+
+
+for threshold in (0.03, 0.04):
+  for branching_factor in (50,55,60):
+    birch_model = Birch(n_clusters=2, threshold=threshold, branching_factor=branching_factor)
+    birch_model.fit(X_scaled.loc[:,'Rainf':'SoilM_100_200cm'])
+    birch_results = birch_model.predict(merged_df.loc[:,'Rainf':'SoilM_100_200cm'])
+    print("Threshold: ", threshold, "Branching factor: ", branching_factor)
+    print(np.unique(birch_results, return_counts=True))
+    print("Accuracy: ", round(accuracy_score(merged_df["pustynia_i_step"], birch_results)*100, 2))
+    print("Precision: ", round(precision_score(merged_df["pustynia_i_step"], birch_results)*100, 2))
+    print("Recall: ", round(recall_score(merged_df["pustynia_i_step"], birch_results)*100, 2))
+    print("F1-score: ", round(f1_score(merged_df["pustynia_i_step"], birch_results)*100, 2))
+    print("- - - - - - - - - - - - - -")
+
+
+# In[21]:
+
+
+birch_model = Birch(n_clusters=2, threshold=0.04, branching_factor=55)
+birch_model.fit(X_scaled.loc[:,'Rainf':'SoilM_100_200cm'])
+birch_results = birch_model.predict(merged_df.loc[:,'Rainf':'SoilM_100_200cm'])
+
+
+# In[22]:
+
+
+merged_df["Cluster"] = birch_results
+
+
+# In[24]:
+
+
+colormap_cluster = get_colormap([0, max(birch_results)], ['darkgreen', "orange"])
+output_notebook()
+show(plot_map(df=merged_df, parameter_name='Cluster', colormap=colormap_cluster, title="Wyniki na zbiorze anotowanym", alpha=0.5))
+
+
+# # Wnioski
+
+# * Grupowanie k-średnich z wykorzystaniem t-SNE nie powiodło się sukcesem ze
+# względu na duży zbiór danych. Po zastosowaniu algorytmu dla różnych wartości preplexity oraz learning_rate algorytm nie rozdzielił danych, co uniemożliwiło grupowanie.
+# * Grupowanie hierarchiczne dla parametrów linkage: "ward", "average", "complete" nie powiodło się ponieważ na Colabie jest zbyt mała pamięć RAM i za każdym razem usuwało sesję. Grupowanie to zadziałało tylko dla linkage = "single" jednak wyniki na zbiorze anotowanym były gorsze od tych powyżej.
```

Comment:
W twoim pliku widzę wyniki klastrowania dla linkage='ward'. Czyli rozumiem, że to jednak dla "ward" się przeliczyło (choć wrzuciło do drugiego klastra tylko jeden elment), a nie dla "single"? 😉 Bo w notatniku nie ma wyników klastrowania dla linkage='single'....